File: src/exercises/comprehensions.py
def sum_evens(numbers: list[int]) -> int:
    return sum(x for x in numbers if not x % 2)


def square_odds(numbers: list[int]) -> list[int]:
    return [x**2 for x in numbers if x % 2]


def word_lengths(words: list[str]) -> dict[str, int]:
    return {w: len(w) for w in words}

File: typescript/src/exercises/warm_up/comprehensions.ts
/**
 * Python list/dict comprehensions map to TypeScript array methods
 * (filter/map/reduce) and `Object.fromEntries`. Functions are named in
 * camelCase to stay idiomatic; the file path mirrors the Python module.
 */

export function sumEvens(numbers: number[]): number {
  return numbers.filter((x) => x % 2 === 0).reduce((sum, x) => sum + x, 0);
}

export function squareOdds(numbers: number[]): number[] {
  return numbers.filter((x) => x % 2 !== 0).map((x) => x ** 2);
}

export function wordLengths(words: string[]): Record<string, number> {
  return Object.fromEntries(words.map((w): [string, number] => [w, w.length]));
}
